### ISSUE TITLE
Problems with `curve` method from `SpaceCurves.m2`

### ISSUE THREAD

**Issue by mahrud** [Macaulay2/M2#937]
One issue is that method `curve` from `SpaceCurves.m2` outputs different types. Here's a sample:
```
needsPackage "SpaceCurves";
curve(2,0)
-- o3 = [...some large ideal...]
-- o3 : Ideal of kk[x , x , x , x ]
--                  0   1   2   3
curve(3,0)
-- o4: [...some large curve...]
-- o4 : Curve
```

The problem involves this function:
https://github.com/Macaulay2/M2/blob/2835cc1b4bd7bb85d030b1985e18e8f10c3abae3/M2/Macaulay2/packages/SpaceCurves.m2#L570-L575

Another issue is the following error:
```
R=ZZ/2[w,x,y,z];
curve(3,0,R)
-- stdio:3:1:(3): error: encountered values for 2 variables, but expected 4
```

And seems to come from this line, when the ideal has degree 5 there aren't enough generators to make a map:
https://github.com/Macaulay2/M2/blob/2835cc1b4bd7bb85d030b1985e18e8f10c3abae3/M2/Macaulay2/packages/SpaceCurves.m2#L105-L108

**Comment by DanGrayson** [comment]
Thanks for the report!

**Comment by mikestillman** [comment]
I've emailed Mengyuan about this. I don't see his handle on github.  It would be good to fix though, but as he is the principal author, I would like his input.

**Comment by mikestillman** [comment]
I've fixed the second issue above, by checking the number of generators and giving a (hopefully) useful error message.  The first issue (curve returning two different types), I'm going to not fix now, as it is just a bit unclean, and also, Mengyuan mentions that he is working on a version 2 of this package.

Fixed in commit a1dc778cf7a7.  Closing this issue for now.  Please reopen if unhappy with not fixing the first part!
